Content.Shared/Actions/BaseActionComponent.cs

```diff
@@ -120,6 +120,12 @@ public abstract partial class BaseActionComponent : Component
     /// </summary>
     [ViewVariables] public EntityUid? AttachedEntity;
 
+    /// <summary>
+    ///     The preferred entity to raise an action event on.
+    /// </summary>
+    [DataField("preferredEntity"), ViewVariables]
+    public EntityUid PreferredEntity;
```

Comment:
See my other comment about the event target. This should either not exist, or be a nullable field that doesn't get automatically set. It should probably also have a clearer name, i.e., `EventTarget` or something like that.

Reply:
I can rename it but I don't understand why it shouldn't exist or it should be nullable because it has to be raised.

Reply:
From what I saw, currently the field isn't ever actually used to store information, its always set directly before being used, so it could just as well be a local variable.

If you want to change it to be an actual datafield that's configurable per action, then it should be optional/nullable.